C/plugins/south/dht11/plugin.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * FogLAMP south plugin.
+ *
+ * Copyright (c) 2018 OSisoft, LLC
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Amandeep Singh Arora
+ */
+#include <dht11.h>
+#include <plugin_api.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <strings.h>
+#include <string>
+#include <logger.h>
+#include <plugin_exception.h>
+#include <config_category.h>
+#include <rapidjson/document.h>
+
+using namespace std;
+
+/**
+ * Default configuration
+ */
+#define CONFIG  "{\"plugin\" : { \"description\" : \"DHT11 plugin\", " \
+                        "\"type\" : \"string\", \"default\" : \"foglamp-dht11\" }, " \
+                "\"asset\" : { \"description\" : \"Asset name\", "\
+                        "\"type\" : \"string\", \"default\" : \"dht11\" }, " \
+                "\"pin\" : { \"description\" : \"RPi pin to which DHT11 is attached\", " \
+                        "\"type\" : \"integer\", \"default\" : \"7\" } "\
+                "}"
+
+
+/**
+ * The DHT11 plugin interface
+ */
+extern "C" {
+
+/**
+ * The plugin information structure
+ */
+static PLUGIN_INFORMATION info = {
+	"DHT11",                  // Name
+	"1.0.0",                  // Version
```

Comment:
this should match the SOVERSION? it may get importance once we have debian packaging versions

Reply:
Yes, will deal with it when we do so.